physics/ephemeris.hpp

```diff
@@ -171,20 +172,20 @@ class Ephemeris {
       not_null<MassiveBody const*> body) const;
 
   // Returns true if at least one of the trajectories is empty.
-  virtual bool empty() const EXCLUDES(lock_);
+  virtual bool empty() const;
 
   // The maximum of the |t_min|s of the trajectories.
   virtual Instant t_min() const EXCLUDES(lock_);
   // The mimimum of the |t_max|s of the trajectories.
-  virtual Instant t_max() const EXCLUDES(lock_);
+  virtual Instant t_max() const;
```

Comment:
Comment: the values of `t_min()` and `t_max()` are only affected by calls to `Prolong` and `ForgetBefore`; while `lock_` is held, their value cannot be changed by another thread.

In the absence of calls to `ForgetBefore`, `t_max()` can only increase, and `t_min()` can only decrease.

This effectively makes `ForgetBefore` hostile to operating on a trajectory from another thread.

Perhaps we should get rid of it; it makes things very messy.